### DISCUSSION TITLE
Can we re-executed entry events when restarting the interpreter from a saved state?

### DISCUSSION THREAD

**Discussion by huan** [statelyai/xstate#3690] [Q&A]
> this is expected behavior, especially for the scenarios when initial states have actions that must be executed.

I agree with you about the above and I'm trying to find a way to restore a state to the machine/interpreter with some actions _"that must be executed"_.

My use case is that once an `EVENT` happens and the `State` transitions to another, the new `State` will want to use `entry` to execute `actions` and deal with the `EVENT`.

If I restart the interpreter with a saved state, my machine needs the `EVENT` to be executed in the `entry`, or the logic will not be expected anymore.

@davidkpiano Is there any way to fulfill the above use case with the latest XState?

_Originally posted by @huan in https://github.com/statelyai/xstate/issues/194#issuecomment-1120393276_

**Comment by Andarist** [comment]
Could you provide some real-life (concrete) use cases for this?

**Comment by bombillazo** [comment]
Waiting on this as well, I need for `start` to run the entry actions defined on the state passed as the initial state.

**Comment by Andarist** [comment]
> Could you provide some real-life (concrete) use cases for this?
